src/App.jsx

```diff
@@ -1,3 +1,95 @@
-export default function App() {
-  return <h1 className="text-3xl font-bold underline bg-blue-500">React exercises</h1>;
-}
+import { useState, useEffect } from "react";
+import Dice from "./Components/Dice";
+import { nanoid } from "nanoid";
+import Confetti from "react-confetti";
+
+const App = () => {
+  const [dice, setDice] = useState(generateNewDice);
+  const [tenzies, setTenzies] = useState(false);
+
+  function generateNewDice() {
+    const newDice = [];
+    for (let i = 0; i < 10; i++) {
+      newDice.push({
+        id: nanoid(),
+        value: Math.ceil(Math.random() * 6),
+        isHeld: false,
+      });
+    }
+    return newDice;
+  }
+
+  const rollNewDice = () => {
+    if (!tenzies) {
+      setDice(
+        dice.map((die) => {
+          return die.isHeld
+            ? die
+            : {
+                id: nanoid(),
+                value: Math.ceil(Math.random() * 6),
+                isHeld: false,
+              };
+        })
+      );
+    } else {
+      setDice(generateNewDice);
+      setTenzies(false);
+    }
+  };
+
+  const handleHold = (id) => {
+    setDice(
+      dice.map((die) => {
+        return die.id === id ? { ...die, isHeld: !die.isHeld } : die;
+      })
+    );
+  };
+
+  useEffect(() => {
+    const allHeld = dice.every((die) => die.isHeld);
+    const allSame = dice.every((die) => die.value === dice[0].value);
+
+    if (allHeld && allSame) {
+      setTenzies(true);
+    }
+  }, [dice]);
+
+  const diceElement = dice.map((die) => (
```

Comment:
Also here you should implement the same thing. remove the brackets from the die argument